polkadot/runtime/westend/src/weights/frame_system_extensions.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Autogenerated weights for `frame_system_extensions`
+//!
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
+//! DATE: 2023-12-20, STEPS: `2`, REPEAT: `2`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! WORST CASE MAP SIZE: `1000000`
+//! HOSTNAME: `gleipnir`, CPU: `AMD Ryzen 9 7900X 12-Core Processor`
+//! WASM-EXECUTION: `Compiled`, CHAIN: `Some("westend-dev")`, DB CACHE: 1024
+
+// Executed Command:
+// ./target/release/polkadot
+// benchmark
+// pallet
+// --steps=2
+// --repeat=2
+// --extrinsic=*
+// --wasm-execution=compiled
+// --heap-pages=4096
+// --pallet=frame-system-extensions
+// --chain=westend-dev
+// --output=./polkadot/runtime/westend/src/weights/
+// --header=./polkadot/file_header.txt
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+#![allow(missing_docs)]
+
+use frame_support::{traits::Get, weights::Weight};
+use core::marker::PhantomData;
+
+/// Weight functions for `frame_system_extensions`.
+pub struct WeightInfo<T>(PhantomData<T>);
+impl<T: frame_system::Config> frame_system::ExtensionsWeightInfo for WeightInfo<T> {
```

Comment:
How is this stuff generated? Can external devs easily learn this? the

 fact that the new tx-extension is weight-aware seems like an important detail, and should be documented possibly in the ref-doc.

Reply:
> fact that the new tx-extension is weight-aware seems like an important detail, and should be documented possibly in the ref-doc

Agreed, the doc work is scoped.

> How is this stuff generated? Can external devs easily learn this?

In order for them to properly work with the current runtime benchmarking system, weights need to belong to a pallet and be exposed through a pallet type, namely a `WeightInfo`. Specifically for `frame_system`, the extensions are treated separately throughout the whole pallet and have this special naming convention, in case people want separate weights for their regular `frame_system` extrinsics and then their extensions, if they use them.

For other extensions it's usually simpler, just benchmark them in the pallet they belong to (e.g. `CheckSudo` and `pallet-sudo`).